src/injected/scan-incomplete-warning-detector.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { ScanIncompleteWarningId } from 'common/types/scan-incomplete-warnings';
+import { IframeDetector } from './iframe-detector';
+
+export interface CrossOriginPermissionDetector {
+    hasCrossOriginPermissions(): boolean;
+}
+
+export class ScanIncompleteWarningDetector {
+    constructor(private iframeDetector: IframeDetector, private crossOriginPermissionDetector: CrossOriginPermissionDetector) {}
+
+    public detectScanIncompleteWarnings = () => {
+        const warnings: ScanIncompleteWarningId[] = [];
+        if (this.iframeDetector.hasIframes() && !this.crossOriginPermissionDetector.hasCrossOriginPermissions()) {
+            warnings.push('missing-required-cross-origin-permissions');
```

Comment:
Can we store this as a variable instead and reference it here and in the type declaration?

Reply:
It would be technically possible to do this in `scan-incomplete-warnings`:

```ts
export const missingRequiredCrossOriginPermissions = 'missing-required-cross-origin-permissions' as const;

// This is a string-enum, but we only have one type right now
export type ScanIncompleteWarningId = typeof missingRequiredCrossOriginPermissions;
```

...but I find this less readable than just using the string directly and relying on the string-enum type to enforce the use of specific string values, personally. @smoralesd / @lisli1 , care to provide a tiebreaker opinion? :)

Reply:
I also prefer the use of the string directly, it reads better... the only downside I can see (I have actually seen it before) shows up if we decide to change the values of the string enum (e.g. in the case of a typo). I think in that case, VSCode rename won't work. I still prefer it this way though.